### PR TITLE
lua_api: Document minetest.features

### PR DESCRIPTION
**Pull request by SmallJoker**
Fixes issue #4823

### CODE REVIEW

File: doc/lua_api.txt [hunk 1]
```diff
@@ -1934,12 +1934,31 @@ Helper functions
...
+        use_texture_alpha = true,
+     -- ^ Use transparent information of textures aswell 
+        no_legacy_abms = true,
+     -- ^ Tree and grass ABMs are no longer contain on C++ side
+        texture_names_parens = true,
+     -- ^ Texture grouping is possible using parentheses
+        area_store_custom_ids = true,
+     -- ^ Unique Area ID for AreaStore:insert_area
+        add_entity_with_staticdata = true,
+     -- ^ Entities can contain staticdata
```
Comment (kaeza) [inline doc/lua_api.txt]: add_entity supports passing initial staticdata to on_activate

Comment (SmallJoker) [inline doc/lua_api.txt]: Thanks, applied.

File: doc/lua_api.txt [hunk 2]
```diff
@@ -1934,12 +1934,31 @@ Helper functions
...
+    * Table containing API feature flags:
+    {
+        glasslike_framed = true,
+        nodebox_as_selectionbox = true,
+        chat_send_player_param3 = true,
+        get_all_craft_recipes_works = true,
+        use_texture_alpha = true,
+     -- ^ Use transparent information of textures aswell 
+        no_legacy_abms = true,
+     -- ^ Tree and grass ABMs are no longer contain on C++ side
```
Comment (sfan5) [inline doc/lua_api.txt]: <i>Tree and grass ABMs are no longer done from C++</i>

File: doc/lua_api.txt [hunk 3]
```diff
@@ -1934,12 +1934,31 @@ Helper functions
...
 * `minetest.features`
-    * Table containing API feature flags: `{foo=true, bar=true}`
+    * Table containing API feature flags:
+    {
+        glasslike_framed = true,
+        nodebox_as_selectionbox = true,
+        chat_send_player_param3 = true,
+        get_all_craft_recipes_works = true,
+        use_texture_alpha = true,
+     -- ^ Use transparent information of textures aswell
```
Comment (sfan5) [inline doc/lua_api.txt]: <i>The transparency channel of textures can be used optionally</i>

File: doc/lua_api.txt (hunk 2 above)
Comment (SmallJoker) [inline doc/lua_api.txt]: This almost sounds like the ABMs are entirely worked out in Lua.

**Comment by SmallJoker** [comment]
Updated the two lines.
----
This gray button feels too much like a "Cancel" one. Sorry for the accidental close :3

**Comment by rubenwardy** [comment]
More spacing  would be nice:

```markdown
* `minetest.features` - Table containing API feature flags:

    {
        glasslike_framed = true,
        nodebox_as_selectionbox = true,
        chat_send_player_param3 = true,
        get_all_craft_recipes_works = true,
        use_texture_alpha = true,
     -- ^ The transparency channel of textures can be used optionally
        no_legacy_abms = true,
     -- ^ Tree and grass ABMs are no longer done from C++
        texture_names_parens = true,
     -- ^ Texture grouping is possible using parentheses
        area_store_custom_ids = true,
     -- ^ Unique Area ID for AreaStore:insert_area
        add_entity_with_staticdata = true,
     -- ^ add_entity supports passing initial staticdata to on_activate
        no_chat_message_prediction = true,
     -- ^ Chat messages are no longer predicted
     }
```

**Comment by SmallJoker** [comment]
@rubenwardy This is quite consistent to other places: [Node boxes](https://github.com/minetest/minetest/blob/master/doc/lua_api.txt#L712-L740)
I'd prefer to see some style rules for the Lua API file. It is modified quite often and by a lot different people.

**Comment by kaeza** [comment]
@SmallJoker Markdown requires a space there, otherwise it takes the block as a single paragraph:

Your version:

* `minetest.features`:
  * Table containing API feature flags:
    {
        glasslike_framed = true,
        nodebox_as_selectionbox = true,
        chat_send_player_param3 = true,
        get_all_craft_recipes_works = true,
        use_texture_alpha = true,
     -- ^ The transparency channel of textures can be used optionally
        no_legacy_abms = true,
     -- ^ Tree and grass ABMs are no longer done from C++
        texture_names_parens = true,
     -- ^ Texture grouping is possible using parentheses
        area_store_custom_ids = true,
     -- ^ Unique Area ID for AreaStore:insert_area
        add_entity_with_staticdata = true,
     -- ^ add_entity supports passing initial staticdata to on_activate
        no_chat_message_prediction = true,
     -- ^ Chat messages are no longer predicted
     }

@rubenwardy's version:

* `minetest.features`: Table containing API feature flags:

    {
        glasslike_framed = true,
        nodebox_as_selectionbox = true,
        chat_send_player_param3 = true,
        get_all_craft_recipes_works = true,
        use_texture_alpha = true,
     -- ^ The transparency channel of textures can be used optionally
        no_legacy_abms = true,
     -- ^ Tree and grass ABMs are no longer done from C++
        texture_names_parens = true,
     -- ^ Texture grouping is possible using parentheses
        area_store_custom_ids = true,
     -- ^ Unique Area ID for AreaStore:insert_area
        add_entity_with_staticdata = true,
     -- ^ add_entity supports passing initial staticdata to on_activate
        no_chat_message_prediction = true,
     -- ^ Chat messages are no longer predicted
     }

My suggestion:

* `minetest.features`: Table containing API feature flags:

      {
          glasslike_framed = true,
          nodebox_as_selectionbox = true,
          chat_send_player_param3 = true,
          get_all_craft_recipes_works = true,
          use_texture_alpha = true,
      --  ^ The transparency channel of textures can be used optionally
          no_legacy_abms = true,
      --  ^ Tree and grass ABMs are no longer done from C++
          texture_names_parens = true,
      --  ^ Texture grouping is possible using parentheses
          area_store_custom_ids = true,
      --  ^ Unique Area ID for AreaStore:insert_area
          add_entity_with_staticdata = true,
      --  ^ add_entity supports passing initial staticdata to on_activate
          no_chat_message_prediction = true,
      --  ^ Chat messages are no longer predicted
      }

Note: The block is indented by 6 spaces: 4 to mark as code block, 2 to mark as being part of the list item.

**Comment by SmallJoker** [comment]
@kaeza Thanks for the explanation. The formatting should look correct now :)

**Comment by SmallJoker** [comment]
@sfan5 Could you please renew your review?

**Pull request merged**